nifi-nar-bundles/nifi-amqp-bundle/nifi-amqp-processors/src/main/java/org/apache/nifi/amqp/processors/PublishAMQP.java

```diff
@@ -0,0 +1,219 @@
+
+            try {
+                this.targetResource.publish(messageContent, amqpProperties);
+                processSession.transfer(flowFile, REL_SUCCESS);
+            } catch (Exception e) {
+                processSession.transfer(processSession.penalize(flowFile), REL_FAILURE);
+                processSession.getProvenanceReporter().receive(flowFile,
+                        this.amqpConnection.toString() + "/E:" + context.getProperty(EXCHANGE).getValue() + "/RK:"
+                                + context.getProperty(ROUNTING_KEY).getValue());
+                this.getLogger().error("Failed while sending message to AMQP via " + this.targetResource, e);
+                context.yield();
+            }
+        }
+    }
+
+    /**
+     *
+     */
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propertyDescriptors;
+    }
+
+    /**
+     *
+     */
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    /**
+     * Will create an instance of {@link AMQPPublisher}
+     */
+    @Override
+    protected AMQPPublisher finishBuildingTargetResource(ProcessContext context) {
+        String exchangeName = context.getProperty(EXCHANGE).getValue();
+        String routingKey = context.getProperty(ROUNTING_KEY).getValue();
+        return new AMQPPublisher(this.amqpConnection, exchangeName, routingKey, this.getLogger());
+    }
+
+    /**
+     * Extracts contents of the {@link FlowFile} as byte array.
+     */
+    private byte[] extractMessage(FlowFile flowFile, ProcessSession session){
+        final byte[] messageContent = new byte[(int) flowFile.getSize()];
+        session.read(flowFile, new InputStreamCallback() {
+            @Override
+            public void process(final InputStream in) throws IOException {
+                StreamUtils.fillBuffer(in, messageContent, true);
+            }
+        });
+        return messageContent;
+    }
+
+    /**
+     * Extracts AMQP properties from the {@link FlowFile} attributes. Attributes
+     * extracted from {@link FlowFile} are considered candidates for AMQP
+     * properties if their names are prefixed with
+     * {@link AMQPUtils#AMQP_PROP_PREFIX} (e.g., amqp$contentType=text/xml)
+     */
+    private BasicProperties extractAmqpPropertiesFromFlowFile(FlowFile flowFile) {
+        Map<String, String> attributes = flowFile.getAttributes();
+        AMQP.BasicProperties.Builder builder = new AMQP.BasicProperties.Builder();
+        for (Entry<String, String> attributeEntry : attributes.entrySet()) {
+            if (attributeEntry.getKey().startsWith(AMQPUtils.AMQP_PROP_PREFIX)) {
+                String amqpPropName = attributeEntry.getKey().split("\\" + AMQPUtils.AMQP_PROP_DELIMITER)[1];
+                String amqpPropValue = attributeEntry.getValue();
+                System.out.println(amqpPropertyNames);
```

Comment:
I think you forgot to remove this? Or change to debug level log message perhaps?